src/reducers/folder.js

```diff
@@ -20,6 +20,8 @@ export const files = (state = [], action) => {
         ...state,
         action.file
       ]
+    case TRASH_FILE_SUCCESS:
+      return state.map(file => file.id === action.file.id ? action.file : file)
```

Comment:
Yes. After a file has been trashed server side, `action.file.dir_id`is going to be different from `file.dir_id`. I suppose the same logic could be used for file moving, renaming, etc.
Does that make sense?